pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -0,0 +1,387 @@
+func (Plugin) OnInboundCluster(env model.Environment, node model.Proxy, service *model.Service,
+	servicePort *model.Port, cluster *xdsapi.Cluster) {
+}
+
+// OnOutboundRouteConfiguration implements the Plugin interface method.
+func (Plugin) OnOutboundRouteConfiguration(in *plugin.InputParams, route *xdsapi.RouteConfiguration) {
+}
+
+// OnInboundRouteConfiguration implements the Plugin interface method.
+func (Plugin) OnInboundRouteConfiguration(in *plugin.InputParams, route *xdsapi.RouteConfiguration) {
+}
+
+// OnOutboundCluster implements the Plugin interface method.
+func (Plugin) OnOutboundCluster(env model.Environment, node model.Proxy, service *model.Service,
+	servicePort *model.Port, cluster *xdsapi.Cluster) {
+}
+
+func isRbacEnabled(store model.IstioConfigStore) (bool, error) {
+	rbacConfigs, err := store.List(model.RbacConfig.Type, kube.IstioNamespace)
+	if err != nil {
+		return false, fmt.Errorf("failed to get rbacConfig: %v", err)
+	}
+	if len(rbacConfigs) != 1 {
+		return false, fmt.Errorf("found %d rbacConfigs, expecting only 1 at most", len(rbacConfigs))
+	}
+	configProto := rbacConfigs[0].Spec.(*rbacproto.RbacConfig)
+	// TODO(yangminzhu): Supports ON_WITH_INCLUSION and ON_WITH_EXCLUSION.
+	if configProto.Mode != rbacproto.RbacConfig_ON {
+		log.Debugf("rbac plugin disabled by rbacConfig: %v", *configProto)
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// buildHTTPFilter builds a http filter that enforces the rbac rules for the specified service in
+// the sidecar proxy.
+func buildHTTPFilter(hostName model.Hostname, store model.IstioConfigStore) (*http_conn.HttpFilter, error) {
+	service := string(hostName)
+	split := strings.Split(service, ".")
+	if len(split) < 2 {
+		return nil, fmt.Errorf("failed to extract namespace from service: %s", service)
+	}
+	namespace := split[1]
+
+	roles, err := store.List(model.ServiceRole.Type, namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ServiceRoles in namespace %s: %v", namespace, err)
+	}
+
+	bindings, err := store.List(model.ServiceRoleBinding.Type, namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ServiceRoleBinding in namespace %s: %v", namespace, err)
+	}
+
+	log.Debugf("%s: converting RBAC rules to proxy config", RbacFilterName)
+	config, err := convertRbacRulesToFilterConfig(service, roles, bindings)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert RBAC rules to filter config: %v", err)
+	}
+
+	return &http_conn.HttpFilter{
+		Name:   RbacFilterName,
+		Config: util.MessageToStruct(config),
+	}, nil
+}
+
+// convertRbacRulesToFilterConfig converts the current RBAC rules in service mesh to proxy config
+// for the specified service.
+func convertRbacRulesToFilterConfig(service string, roles []model.Config, bindings []model.Config) (*policyproto.RBAC, error) {
+	// roleToBinding maps ServiceRole name to a list of ServiceRoleBindings.
+	roleToBinding := map[string][]*rbacproto.ServiceRoleBinding{}
```

Comment:
Can you add some example generated config and more comments on what is generated ? 

Does each inbound listener gets all the rbac bindings in the mesh ? A subset ? 
I assume only roles and policies targeted to the service are included.

If this is the case - can we use a better data structure - like a map, instead if iterating trough the full list every single time ?

Reply:
I think this pre-processing should be cached somehow. We can think about it for the next PRs.